### ISSUE TITLE
Proper handling of validation errors

### ISSUE THREAD

**Issue by ianmuchyri** [absmach/magistrala-ui#5]
Since most transport errors are a result of user input errors, I would suggest that these errors be encoded as status codes, and handled on the client side. The client-side would use the javascript `fetch` function to interrupt the default form submission. 
Lets take` createUser` as an example: 
Create an HTTP request using the `fetch`, then handle the response based on the various status codes you expect.
### Scenarios

1. Successful creation of user

- In the endpoints.go file we return uiRes with status code 201 (StatusCreated)

`func createUserEndpoint(svc ui.Service) endpoint.Endpoint {
	return func(_ context.Context, request interface{}) (interface{}, error) {
		req := request.(createUserReq)

		if err := req.validate(); err != nil {
			return nil, err
		}
		if err := svc.CreateUsers(req.token, req.User); err != nil {
			return nil, err
		}

		return uiRes{
			code:    http.StatusCreated,
		}, nil
	}
}`

- In the submitUserForm javascript function we would handle the case `response.status===201` and return a `banner` for successful user creation before reloading the page `window.location.reload();`

3. User already exists in the database
- In the endpoints.go file the function would `return nil, err` since we would have conflict error.
- In transport.go file `encodeError `function, you encode the error with 409 (StatusConflict)
- In the javascript functionn you handle `response.status===409` and return a error message for conflict error
5. User input data is incorrect (eg invalid email or password)
- In requests.go file we will validate the struct values and return specific errors based on the validation error. These errors will be handled in the encodeError function in `transport.go `and return `statuscodes (e.g 400 (bad request) or 403(Forbiden))`, and include a `custom header` in the response that conveys the particular validation error. 
- In the javascript function we handle the status code and fetch the error header from the body of the status code. This error is what we display to the user.

**Comment by rodneyosodo** [comment]
For example, if a user tries to add someone to a domain where they are not a member, the UI does not handle this error. The API works fine. Logs from the UI container

```bash
{"level":"info","message":"Method members for domain 20adc8ca-a9c6-4138-a000-d53d6ca0e82e took 10.035372ms to complete without errors.","ts":"2023-12-18T09:48:30.701424722Z"}
{"level":"info","message":"Method get_entities took 3.057889ms to complete without errors.","ts":"2023-12-18T09:48:30.804131483Z"}
{"level":"warn","message":"Method assign_member for domain 20adc8ca-a9c6-4138-a000-d53d6ca0e82e took 4.935097ms to complete with error: Status: Conflict: failed to add policies : failed to add policies.","ts":"2023-12-18T09:48:37.303181861Z"}
```

**Comment by dborovcanin** [comment]
Resolved by #1.
